fdbclient/MultiVersionTransaction.actor.cpp

```diff
@@ -895,22 +895,43 @@ MultiVersionDatabase::MultiVersionDatabase(MultiVersionApi* api,
 
 		api->runOnExternalClients(threadIdx, [this](Reference<ClientInfo> client) { dbState->addClient(client); });
 
-		if (!externalClientsInitialized.test_and_set()) {
-			api->runOnExternalClientsAllThreads([&clusterFilePath](Reference<ClientInfo> client) {
-				// This creates a database to initialize some client state on the external library
-				// We only do this on 6.2+ clients to avoid some bugs associated with older versions
-				// This deletes the new database immediately to discard its connections
-				if (client->protocolVersion.hasCloseUnusedConnection()) {
-					Reference<IDatabase> newDb = client->api->createDatabase(clusterFilePath.c_str());
+		api->runOnExternalClientsAllThreads([&clusterFilePath](Reference<ClientInfo> client) {
+			// This creates a database to initialize some client state on the external library
+			// We only do this on 6.2+ clients to avoid some bugs associated with older versions
+			// This deletes the new database immediately to discard its connections
+			if (client->protocolVersion.hasCloseUnusedConnection() && !client->initialized) {
+				MutexHolder holder(client->initializationMutex);
```

Comment:
I'm debating removing this mutex. I don't think it's needed for correctness, but it's instead used as an optimization to prevent multiple simultaneous attempts to initialize a client when some of them may be unnecessary. Adding a mutex creates a little bit of risk for what seems like a fairly minor optimization.

Reply:
what is the risk of this mutex?

Reply:
I was just thinking of the typical risks of mutexes, such as deadlocks. There are other locks that exist in this code, and while I was able to convince myself that there shouldn't be an issue with that here, it's still possible that I missed something or that the code changes in a way that causes an issue. The benefit of having it seemed fairly low, so I'm not sure it's worth it.

Reply:
I see, that makes sense to me.